Tests/RegexTests/LexTests.swift

```diff
@@ -14,91 +14,7 @@
 import XCTest
 @testable import _StringProcessing
 
-func diagnose(
-  _ input: String,
-  expecting expected: ParseError,
-  _ syntax: SyntaxOptions = .traditional,
-  _ f: (inout Source) throws -> (),
-  file: StaticString = #file,
-  line: UInt = #line
-) {
-  var src = Source(input)
-  do {
-    try f(&src)
-    XCTFail("""
-      Passed, but expected error: \(expected)
-    """, file: file, line: line)
-  } catch let e as Source.LocatedError<ParseError> {
-    guard e.error == expected else {
-      XCTFail("""
-
-        Expected: \(expected)
-        Actual: \(e.error)
-      """, file: file, line: line)
-      return
-    }
-  } catch let e {
-    fatalError("Should be unreachable: \(e)")
-  }
-}
-
 extension RegexTests {
-  func testLexicalAnalysis() {
-    diagnose("a", expecting: .expected("b")) { src in
-      try src.expect("b")
-    }
-
-    diagnose("", expecting: .unexpectedEndOfInput) { src in
-      try src.expectNonEmpty()
-    }
-    diagnose("a", expecting: .unexpectedEndOfInput) { src in
-      try src.expect("a") // Ok
-      try src.expectNonEmpty() // Error
-    }
```

Comment:
Why are we removing these tests?

Reply:
Put them back, except the scalar tests which I moved to ParseTests